### ISSUE TITLE
RFC: Should `Access-Control-Allow-Origin: *` incur negative points?

### ISSUE THREAD

**Issue by u2mejc** [mozilla/http-observatory#358]
TL;DR: IMHO, only a CDN should use `Access-Control-Allow-Origin: *`

From enable-cors.org:
> Cross-Origin Resource Sharing (CORS) is a specification that enables truly open access across domain-boundaries. If you serve public content, please consider using CORS to open it up for universal JavaScript/browser access.

From this description, CORS isn't a security feature, it sidesteps cross-origin resource security by allowing certain domains access. When CORS is _not_ enabled, the default behaivor is for browsers to block non-origin javascript requests. One of the reasons is to protect cookie access which may contain session data saved by the browser. That's very high level and I won't attempt to recreate the awesome work of other security professionals, but a quick google search for `cors security risk` or looking at the issue for the enable-cors.org repo, will give more details: https://github.com/monsur/enable-cors.org/issues/108

People use Mozilla observatory to advise on security readiness of their websites. Unfortunately, not everyone who uses observatory will understand the security consequences of statically setting `Access-Control-Allow-Origin: *` on all server responses, as _recommended_ by the popular google result, enable-cors.org.

Mozilla observatory today gives any site that staticly sets this, zero negative points and a green check for passing. I'd like to recommend that `cross-origin-resource-sharing-implemented-with-universal-access` and `cross-origin-resource-sharing-implemented-with-public-access` have the same negative point weight.

**Comment by april** [comment]
`ACAO: *` alone isn't terrible scary, however, unless they have private data visible publicly. In those cases, they usually use countermeasures like complex, unique URLs.

Note that when making a cross-origin read with only `ACAO: *`, cookies are *not* sent. The requests are completely anonymous. As such, I'm not sure what you mean by "protect cookie access".

**Comment by u2mejc** [comment]
Thank you for your quick response @april !

Your correct, `ACAO: *` alone doesn't allow cookie access, sorry to have typed that quickly. I amended the above to say non-public data. 

The attack vector is implicit authentication, and the problem is the risk that `ACAO: *` is generally accepted safe behavior, purported by a highly ranked page _and_ Mozilla observatory agrees there is no harm.

Example: If a junior IT engineer setups an intr_a_net wiki with sensitive data (and they happened to google CORS and copied the first response), this would allow a external malicious site to load data and potentially upload it to a remote site. That example is boring, let me try again. Imagine your team has your production Solr cluster white listed to your office or simply available on VPN from your machine, and it just happens to include `ACAO: *`, because it came up in your [results](http://laurenthinoul.com/how-to-enable-cors-in-solr/) when searching for solr.

Mozilla Observatory is available both publicly and privately when downloaded. The question is, should a user always receive a gold star (green check in this case) for setting `ACAO: *`, or should there be a gentle reminder that this may divulge some unexpected data when used with implicit authentication?
